### PR TITLE
Move BlockTimestampDriftNanoSecs to GlobalParams

### PR DESCRIPTION
**Pull request by lazynina**

### CODE REVIEW

**Comment by lazynina** [comment]
Current dependencies on/for this PR:
* `feature/proof-of-stake`
  * **PR #1027** <a href="https://app.graphite.dev/github/pr/deso-protocol/core/1027?utm_source=stack-comment-icon" target="_blank">Graphite</a>  👈

This [stack of pull requests](https://stacking.dev/?utm_source=stack-comment) is managed by [Graphite](https://app.graphite.dev/github/pr/deso-protocol/core/1027?utm_source=stack-comment).

File: lib/pos_blockchain.go [hunk 1]
```diff
@@ -507,9 +514,14 @@ func (bc *Blockchain) processOrphanBlockPoS(block *MsgDeSoBlock) error {
 		// As a spam-prevention measure, we just throw away this block and don't store it.
 		return nil
 	}
+	// Get the snapshot global params
+	snapshotGlobalParams, err := utxoView.GetCurrentSnapshotGlobalParamsEntry()
```
Comment (tholonious) [inline lib/pos_blockchain.go:518]: Oof, this one is tricky. You'll want to use the snapshot entry based on the epoch the orphan block falls in (not the current snapshot global params based on the committed tip). I think you need to move the isProperlyFormedBlock into the if/else blocks above.

File: lib/pos_blockchain.go [hunk 2]
```diff
@@ -144,9 +144,16 @@ func (bc *Blockchain) validateAndIndexHeaderPoS(header *MsgDeSoHeader, headerHas
 			header, errors.New("validateAndIndexHeaderPoS: Parent header failed validations"),
 		)
 	}
-
+	utxoView, err := bc.getUtxoViewAtBlockHash(*header.PrevBlockHash)
+	if err != nil {
+		return nil, false, errors.Wrap(err, "validateAndIndexHeaderPoS: Problem getting UtxoView")
+	}
+	snapshotGlobalParams, err := utxoView.GetCurrentSnapshotGlobalParamsEntry()
```
Comment (tholonious) [inline lib/pos_blockchain.go:151]: Add a comment here to elaborate on why we use the snapshot global params. It's to make the drift timestamp check work for orphan blocks.

**Review by tholonious — CHANGES_REQUESTED** [review]

File: lib/pos_blockchain.go (hunk 1 above)
Comment (lazynina) [reply lib/pos_blockchain.go:518]: good catch!

File: lib/pos_blockchain.go [hunk 3]
```diff
@@ -812,14 +842,14 @@ func (bc *Blockchain) isBlockTimestampValidRelativeToParentPoS(header *MsgDeSoHe
 // isProperlyFormedBlockPoS validates the block at a surface level and makes
 // sure that all fields are populated in a valid manner. It does not verify
 // signatures nor validate the blockchain state resulting from the block.
-func (bc *Blockchain) isProperlyFormedBlockPoS(block *MsgDeSoBlock) error {
+func (bc *Blockchain) isProperlyFormedBlockPoS(block *MsgDeSoBlock, globalParams *GlobalParamsEntry) error {
```
Comment (tholonious) [inline lib/pos_blockchain.go:845]:
```suggestion
func (bc *Blockchain) isProperlyFormedBlockPoS(block *MsgDeSoBlock, snapshotGlobalParams *GlobalParamsEntry) error {
```

File: lib/pos_blockchain.go [hunk 4]
```diff
@@ -753,9 +783,9 @@ func (bc *Blockchain) validatePreviouslyIndexedBlockPoS(blockHash *BlockHash) (*
 
 // isValidBlockPoS performs all basic block integrity checks. Any error
 // resulting from this function implies that the block is invalid.
-func (bc *Blockchain) isValidBlockPoS(block *MsgDeSoBlock) error {
+func (bc *Blockchain) isValidBlockPoS(block *MsgDeSoBlock, globalParams *GlobalParamsEntry) error {
```
Comment (tholonious) [inline lib/pos_blockchain.go:786]:
```suggestion
func (bc *Blockchain) isValidBlockPoS(block *MsgDeSoBlock, snapshotGlobalParams *GlobalParamsEntry) error {
```

**Review by tholonious — APPROVED** [review]

**Pull request merged**